src/box/memtx_tree.cc

```diff
@@ -356,7 +357,8 @@ static void
 tree_iterator_free(struct iterator *iterator)
 {
 	struct tree_iterator<USE_HINT> *it = get_tree_iterator<USE_HINT>(iterator);
-	tree_iterator_set_last_tuple(it, NULL);
+	if (it->last.tuple != NULL)
+		tuple_unref(it->last.tuple);
```

Comment:
This and the changes in `tree_iterator_set_last_tuple` belong to the previous patch. Please rebase.

Reply:
Done